notebooks/Shuvomoy - Exploiting sparsity.ipynb

```diff
@@ -771,113 +768,115 @@
-    "        push!(demandedAmount, c, 175)  \n",
+    "        setindex!(demandedAmount, 175, c)  \n",
     "#11\n",
     "    elseif c.p==:tablet && c.d==:DUBAI\n",
-    "        push!(demandedAmount, c,  350) \n",
+    "        setindex!(demandedAmount, 350, c  ) \n",
     "#12\n",
     "    elseif c.p==:laptop && c.d==:DUBAI\n",
-    "        push!(demandedAmount, c, 65)\n",
+    "        setindex!(demandedAmount, 65, c)\n",
     "#13\n",
     "    elseif c.p==:smartphone && c.d==:KUALALUMPUR\n",
-    "        push!(demandedAmount, c, 550)\n",
+    "        setindex!(demandedAmount, 550, c)\n",
     "#14\n",
     "    elseif c.p==:tablet && c.d==:KUALALUMPUR\n",
-    "        push!(demandedAmount, c, 950)\n",
+    "        setindex!(demandedAmount, 950, c)\n",
     "#15\n",
     "    elseif c.p==:laptop && c.d==:KUALALUMPUR\n",
-    "        push!(demandedAmount, c, 185) \n",
+    "        setindex!(demandedAmount, 185, c) \n",
     "#16\n",
     "    elseif c.p==:smartphone && c.d==:HONGKONG\n",
-    "        push!(demandedAmount, c, 200)\n",
+    "        setindex!(demandedAmount, 200, c)\n",
     "#17\n",
     "    elseif c.p==:tablet && c.d==:HONGKONG\n",
-    "        push!(demandedAmount, c, 750)\n",
+    "        setindex!(demandedAmount, 750, c)\n",
     "#18\n",
     "    elseif c.p==:laptop && c.d==:HONGKONG\n",
-    "        push!(demandedAmount, c, 150)\n",
+    "        setindex!(demandedAmount, 150, c)\n",
     "#19\n",
     "    elseif c.p==:smartphone && c.d==:BARCELONA\n",
-    "        push!(demandedAmount, c, 275)\n",
+    "        setindex!(demandedAmount, 275, c)\n",
     "#20\n",
     "    elseif c.p==:tablet && c.d==:BARCELONA\n",
-    "        push!(demandedAmount, c, 600)\n",
+    "        setindex!(demandedAmount, 600, c)\n",
     "#21   \n",
     "    elseif c.p==:laptop && c.d==:BARCELONA\n",
-    "        push!(demandedAmount, c, 200)\n",
+    "        setindex!(demandedAmount, 200, c)\n",
     "    else\n",
-    "        push!(demandedAmount, -1)\n",
+    "        setindex!(demandedAmount, -1)\n",
     "        println(\"Something is wrong index\", i, \"with customers\", c)\n",
     "    end\n",
     "end\n",
     "\n",
+    "\n",
     "costCof = \n",
     "[34; 7; 8; 10; 11; 74; 9; 18; 5; 15; 6; 23; 81; 18; 20; 10; 9; \n",
     " 13; 25; 85; 13; 40; 17; 7; 16; 20; 80; 9; 24; 5; 15; 11; 23; \n",
     "90; 22; 19; 15; 16; 15; 24; 100; 21; 37; 12; 9; 16; 14; \n",
     "88; 9; 28; 13; 17; 8; 32; 100; 18; 28; 15; 18; 16; 30; 102; 15]\n",
     "\n",
-    "# Creating costRoutes dictionary which contains\n",
+    "# Creating costRoutes dictionary which contains the costs of the relevant routes\n",
     "#          ----------\n",
     "costRoutes=Dict{Route, Float64}()\n",
-    "for i in [1:length(routes)]\n",
-    "    push!(costRoutes, routes[i], costCof[i])\n",
+    "for i in collect(1:length(routes))\n",
+    "    setindex!(costRoutes,  costCof[i], routes[i])\n",
```

Comment:
I think costRoutes[costCof[i]] = routes[i] is clearer. Is setindex! and
this are equivalent in terms of efficiency? If so i think I will change the
setindex! to what you suggested.

On Mon, Jan 25, 2016 at 6:25 PM, Miles Lubin <notifications@github.com>
wrote:

> In notebooks/Shuvomoy - Exploiting sparsity.ipynb
> <https://github.com/JuliaOpt/juliaopt-notebooks/pull/17#discussion_r50773757>
> :
>
> >      "#          ----------\n",
> >      "costRoutes=Dict{Route, Float64}()\n",
> > -    "for i in [1:length(routes)]\n",
> > -    "    push!(costRoutes, routes[i], costCof[i])\n",
> > +    "for i in collect(1:length(routes))\n",
> > +    "    setindex!(costRoutes,  costCof[i], routes[i])\n",
>
> Would costRoutes[costCof[i]] = routes[i] be clearer?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JuliaOpt/juliaopt-notebooks/pull/17/files#r50773757>.
>



-- 
Shuvomoy Das Gupta
Ph.D. Student
Systems Control Group
Department of Electrical & Computer Engineering
University of Toronto
http://scg.utoronto.ca/~shuvomoy.dasgupta/